locator/abstract_replication_strategy.cc

```diff
@@ -558,6 +558,18 @@ vnode_effective_replication_map::factory_key vnode_effective_replication_map::ma
     return factory_key(rs->get_type(), rs->get_config_options(), tmptr->get_ring_version());
 }
 
+
+size_t vnode_effective_replication_map::get_replication_factor([[maybe_unused]] dht::token id, const seastar::sstring& datacenter) const {
+    const auto& replication_strategy = get_replication_strategy();
+
+    if (replication_strategy.get_type() == replication_strategy_type::network_topology) {
+        const network_topology_strategy* nrs = static_cast<const network_topology_strategy*>(&replication_strategy);
+        return nrs->get_replication_factor(datacenter);
+    }
```

Comment:
Would be cleaner to add dc-aware get_replication_factor() to abstract_replication_strategy so that abstract things stay abstract, and this does not have to be aware of every impl.

Reply:
On the other hand, it mimics preexisting logic, so you can ignore.